code_challenges/code_wars/remove_parentheses.py

```diff
@@ -0,0 +1,36 @@
+class CharacterRemover:
+    def __init__(self, characters):
+        self.characters = characters
+
+    def item_to_remove(self):
+        self.start_item = "("
+        start_item_index = None
+        self.stop_item = ")"
+        stop_item_index = None
+
+        for i in range(len(self.characters)):
+            if self.characters[i] == self.start_item:
+                start_item_index = i
+            elif self.characters[i] == self.stop_item:
+                stop_item_index = i
+                return start_item_index, stop_item_index
+
+    def remove_item(self):
+        start, stop = self.item_to_remove()
+        self.characters = self.characters[:start] + self.characters[stop + 1 :]
+        if self.start_item in self.characters and self.stop_item in self.characters:
+            self.item_to_remove()
+            self.remove_item()
```

Comment:
This is called **recursion** and is a pretty handy way of solving some problems. It does have some draw backs though. Do you know what they are?

Try your code with the characters `"".join([f"a({n})b" for n in range(1000)])` and `"".join([f"a({n})b" for n in range(3000)])` does that work? If so, try `"ab".join([f"a({n})b" for n in range(10000)])`.

-------------------------

Is line 22 needed?

Reply:
RecursionError: maximum recursion depth exceeded while calling a Python object

Reply:
i thought line 22 was needed because im needing to find the new index of the item to remove